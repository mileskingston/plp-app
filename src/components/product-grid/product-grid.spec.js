import React from 'react';
import renderer from 'react-test-renderer';
import ProductGrid from './product-grid';

describe('Product Grid Component', () => {
  const products = [
    {
      name: 'Product',
      price: [{
        currency: 'USD',
        description: 'L',
        usage: 'Display',
        value: '400.0',
      }],
      thumbnail: '/image/url',
    },
    {
      name: 'Product 2',
      price: [{
        currency: 'USD',
        description: 'L',
        usage: 'Display',
        value: '200.0',
      }],
      thumbnail: '/image-2/url',
    },
  ];

  it('renders component', () => {
    let component = renderer.create(<ProductGrid products={products} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders component when no products', () => {
    let component = renderer.create(<ProductGrid products={[]} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});