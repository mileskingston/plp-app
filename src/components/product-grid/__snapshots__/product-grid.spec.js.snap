// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Grid Component renders component 1`] = `
<div
  className="product-grid"
>
  <div
    className="product"
  >
    <a
      href="/"
    >
      <img
        alt="Product"
        className="product__image"
        src="https://www.demo.salmon-gcp.com/image/url"
      />
      <h2
        className="product__name"
      >
        Product
      </h2>
      <p
        className="product__price"
      >
        $400.0
      </p>
    </a>
  </div>
  <div
    className="product"
  >
    <a
      href="/"
    >
      <img
        alt="Product 2"
        className="product__image"
        src="https://www.demo.salmon-gcp.com/image-2/url"
      />
      <h2
        className="product__name"
      >
        Product 2
      </h2>
      <p
        className="product__price"
      >
        $200.0
      </p>
    </a>
  </div>
</div>
`;

exports[`Product Grid Component renders component when no products 1`] = `
<div
  className="product-grid"
>
  <p>
    Sorry, no products
  </p>
</div>
`;
